tket2/src/passes/chunks.rs

```diff
@@ -225,9 +225,12 @@ struct ChunkInsertResult {
 /// The target of a chunk connection in a reassembled circuit.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, From)]
 enum ConnectionTarget {
-    /// The target is a single node and port.
+    /// The target is a chunk node's input.
```

Comment:
(later you can change `incoming_connections` and `outgoing_connections` to have specific direction, by parameterization, avoiding the panic, but that can be another PR, sure...)

Reply:
Hmmmm, given that, are we sure that it doesn't make more sense to keep this as `Port` and then do `as_incoming().unwrap()`/`as_outgoing().unwrap()` in the two places instead of the `panic`s you've added?

Reply:
Having the explicit variants makes it so any direction bug is detected at construction time, rather than when unwrapping. The panic will only be raised for bugs in the intermediate code, so it's a bit more scoped.

Hopefully this will be cleaned out in #220.

Reply:
Yah ok. As an intermediate step this is indeed slightly better I think. I was only thinking that we'll be kind-of reverting this in #220 but of course, it depends how long it takes us to get round to that P-low. So, fine :)